dataplane/forwarding/fwdaction/actions/select_action_list.go

```diff
@@ -130,12 +140,24 @@ func (*selectActionListBuilder) Build(desc *fwdpb.ActionDesc, ctx *fwdcontext.Co
 		return nil, fmt.Errorf("actions: Unable to find select function %v", s.hash)
 	}
 
+	allZeros := true
 	for _, l := range sal.Select.GetActionLists() {
 		a, err := fwdaction.NewActions(l.GetActions(), ctx)
 		if err != nil {
 			return nil, fmt.Errorf("actions: Unable to create actions %v, err %v", l, err)
 		}
 		s.set = append(s.set, a)
+		if l.GetWeight() != 0 {
+			allZeros = false
+		}
+		s.weights = append(s.weights, int(l.GetWeight()+s.weightSum))
+		s.weightSum += l.GetWeight()
+	}
+	if allZeros {
+		s.weightSum = uint64(len(s.weights))
+		for i := range s.weights {
+			s.weights[i] = i + 1
```

Comment:
Why is it i+1? Shouldn't it be just 1?

Reply:
say the hash func (after mod) values returns in vals [0,4), the weights [1,2,3,4]

so h = 0,  h < 1 => (select idx 0)

Reply:
renamed